conans/client/installer.py

```diff
@@ -120,14 +122,41 @@ def _prepare_sources(self, conanfile, pref, package_layout, conanfile_path, sour
             logger.debug("BUILD: Copied to %s", build_folder)
             logger.debug("BUILD: Files copied %s", ",".join(os.listdir(build_folder)))
 
-    def _build(self, conanfile, pref):
+    def _build(self, conanfile, pref, node):
         # Read generators from conanfile and generate the needed files
         logger.info("GENERATORS: Writing generators")
         write_generators(conanfile, conanfile.build_folder, self._output)
 
         logger.info("TOOLCHAIN: Writing toolchain")
         write_toolchain(conanfile, conanfile.build_folder, self._output)
 
+        # I need the shims for the executables in the build-requires (only two-profiles approach)
+        logger.info("SHIMS: Writing shims for build-requires")
```

Comment:
Most likely something we want to make opt-in, I imagine this can break in multiple different ways.

Reply:
If it is a generator, then recipes that want to use it, can just add the generator?

Reply:
I would say a _global config variable_, it will not only generate the wrappers, but it will also stop populating the environment with information from the _build_ context. 

If we use a config variable we don't need to modify the recipes themselves.